src/services/embedding/client.py

```
@@ -91,25 +98,38 @@ async def embed(self, texts: List[str]) -> List[List[float]]:
+            current_loop = None
+
+        if current_loop is None:
+            # No running loop, safe to create a new one
             return asyncio.run(self.embed(texts))
 
+        if current_loop == self._init_loop:
+            # Called sync from within the init loop - this would block
+            raise RuntimeError(
+                "embed_sync() cannot be called from within the same running event loop. "
+                "Use 'await embed()' instead."
+            )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)